exercises/protein-translation/example.cpp

```diff
@@ -0,0 +1,29 @@
+#include "protein_translation.h"
+#include <vector>
+#include <string>
+#include <algorithm>
+#include <map>
+
+using namespace std;
+
+vector<string> protein_translation::proteins(string rna) {
+
+    map<string, string> dic;
+    dic["AUG"] = "Methionine";
+    dic["UUU"] = dic["UUC"] = "Phenylalanine";
+    dic["UUA"] = dic["UUG"] = "Leucine";
+    dic["UCU"] = dic["UCC"] = dic["UCA"] = dic["UCG"] = "Serine";
+    dic["UAU"] = dic["UAC"] = "Tyrosine";
+    dic["UGU"] = dic["UGC"] = "Cysteine";
+    dic["UGG"] = "Tryptophan";
+    dic["UAA"] = dic["UAG"] = dic["UGA"] = "STOP";
+
+    vector<string> polypeptide;
+    for (int i = 0; i < (int)rna.size()-2; i += 3) {
+        string codon = rna.substr(i, 3);
+        string protein = dic[codon];
```

Comment:
The tests for some reason don't check, but this will cause problems if the value in the map doesn't exist. You want to use `dic.find(codon)` instead.

Reply:
We will only get an empty string "", aren't we?

Reply:
With this solution the empty string would be included in the output, which is probably not the desired behavior.. For some reason the tests don't check the error case where the string doesn't exist, I'd assume we'd treat it as either a "STOP" or as an error.

Also to be clear, this line will also add an entry to the map with key = `codon` and the value = the empty string. It's a quirk of c++ maps that you need to be careful with.

(I will probably just fix this myself in a few minutes and push the change, everything else looks good)